### PR TITLE
feat: Delete node functionality, Open node admin dashboard, custom popup for messages

### PR DESCRIPTION
**Pull request by alenmestrov**
**Summary**  
This PR introduces the ability to delete nodes and open an admin dashboard for running nodes. The admin dashboard can be accessed from the tray or when a specific node is selected in the UI, but it is only available when the node is running. Additionally, a custom popup has been implemented to notify users when they attempt to perform an action that cannot be completed due to the node's current state.

**Changes**  
- **Node Deletion**: Implemented functionality to delete a node from the application. This removes the node and its associated data from the system.
- **Admin Dashboard Access**: Added the ability to open an admin dashboard for a running node:
  - The dashboard can be accessed via the tray when a node is selected.
  - It is also accessible from the main UI when interacting with a running node.
  - The admin dashboard is only available when the node is running, and the user gets a notification in a popup if he tries opening it while the node is not running
- **Custom Popup Notifications**: Created a custom popup to inform users when they try to perform an action (like opening the admin dashboard) that is not allowed due to the current state of the node (e.g., when the node is stopped).
- **Tray Integration**: Integrated the admin dashboard into the tray menu, ensuring that the dashboard is only selectable when the node is running.

**Testing**  
- Verified that nodes can be successfully deleted, with all associated data being removed.
- Tested that the admin dashboard option is only available in the tray or UI when a node is running.
- Checked that attempting to open the admin dashboard for a stopped node triggers the custom popup with the appropriate message.
- Confirmed that the tray dynamically updates to reflect the availability of the admin dashboard and deletion options based on the node's status.
- Ensured that the custom popup is displayed when any action is attempted that is restricted due to the node's current state.

**How to Test**  
1. Run the following commands:
   ```bash
   pnpm install
   pnpm run tauri dev
   ```
2. Initialize one or more nodes and verify that:
   - You can delete a node from UI, and its data is removed.
   - The node data should be removed from `/Users/<user>/Library/Application Support/calimero.node.manager/nodes` directory
   - The admin dashboard option is only available when the node is running (both in the tray and UI).
   - Clicking on the admin dashboard when the node is running opens the dashboard.
   - Attempting to open the admin dashboard for a stopped node shows the custom popup notification.
3. Use the tray to interact with the nodes, confirming the availability of the admin dashboard option based on the node's status.
4. Ensure that the custom popup appears with the correct messaging when any action that depends on the node's state is unavailable.

### CODE REVIEW

File: src-tauri/src/operations/node_operations.rs [hunk 1]
```diff
@@ -430,3 +430,75 @@ pub fn send_input_to_node(
...
+
+    Ok(OperationResult {
+        success: true,
+        message: format!("Node '{}' has been deleted successfully.", node_name),
+    })
+}
+
+pub fn open_admin_dashboard(app_handle: AppHandle, node_name: String) -> Result<OperationResult> {
+    let config = get_node_ports(&node_name, &app_handle)?;
+    let url = format!("http://localhost:{}/admin-dashboard", config.server_port);
```
Comment (frdomovic) [inline src-tauri/src/operations/node_operations.rs:485]: Update this so it knows if its http or https

File: src/components/common/MessagePopup.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import PopupWrapper from './PopupWrapper';
+import Button from './Button';
+import styled from 'styled-components';
+
+interface MessagePopupProps {
+  isOpen: boolean;
+  onClose: () => void;
+  message: string;
+  title: string;
+  type?: 'info' | 'warning' | 'error';
```
Comment (frdomovic) [inline src/components/common/MessagePopup.tsx:11]: Optional request.
extract to separate type 
```suggestion
  type?: PopupType; 
```
type PopupType = 'info' | 'warning' | 'error';

Comment (frdomovic) [reply src/components/common/MessagePopup.tsx:11]: or maybe enum

File: src/components/node/DeleteNode.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      setErrorMessage('An unexpected error occurred while deleting the node.');
+    }
+  };
+
+  return (
+    <DeleteNodeContainer>
+      <DeleteNodeText>Are you sure you want to delete this node?</DeleteNodeText>
+      {errorMessage && <ErrorMessage>{errorMessage}</ErrorMessage>}
+      <Button onClick={onCancel} variant="stop">Cancel</Button>
+      <Button onClick={handleDelete} variant="delete">Confirm Delete</Button>
```
Comment (frdomovic) [inline src/components/node/DeleteNode.tsx:34]: Add disabled on this with isLoading

File: src/components/node/DeleteNode.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  handleDeleteNode: () => Promise<CommandResponse>;
+  onCancel: () => void;
+  onDeleteSuccess: () => void;
+}
+
+const DeleteNode: React.FC<DeleteNodeProps> = ({ handleDeleteNode, onCancel, onDeleteSuccess }) => {
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const handleDelete = async () => {
+    try {
```
Comment (frdomovic) [inline src/components/node/DeleteNode.tsx:16]:
```suggestion
    setIsLoading(true); // disable delete button
    try {
    .....
    setIsLoading(false); // process finished
```

Can't test yet but if this component is not popup and stays open it should have this disable options

File: src/components/node/NodeControls.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import React, { useState, useEffect, useRef } from 'react';
 import { invoke } from '@tauri-apps/api/tauri';
 import { listen, UnlistenFn } from '@tauri-apps/api/event';
+// import { dialog } from '@tauri-apps/api';
```
Comment (frdomovic) [inline src/components/node/NodeControls.tsx:4]: is this WIP?

File: src/components/node/NodeControls.tsx [hunk 6]
```diff
@@ -25,6 +27,7 @@ const NodeControls: React.FC<NodeControlsProps> = ({ selectedNode, handleNodeSta
   const [input, setInput] = useState<string>('');
   const [isRunning, setIsRunning] = useState<boolean>(selectedNode.is_running);
   const outputRef = useRef<HTMLPreElement>(null);
+  const [messagePopup, setMessagePopup] = useState({ isOpen: false, message: '', title: '', type: 'info' as const });
```
Comment (frdomovic) [inline src/components/node/NodeControls.tsx:30]: extract to type useState<PopupProps> - or something like this. Use it other components if they are using the same messagePopup structure.
Don't understand why "info as const" ?

File: src/components/common/MessagePopup.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+const MessagePopup: React.FC<MessagePopupProps> = ({ isOpen, onClose, message, title, type = 'info' }) => {
+  return (
+    <PopupWrapper isOpen={isOpen} onClose={onClose}>
+      <MessageContent>
+        <MessageTitle>{title}</MessageTitle>
+        <MessageText>{message}</MessageText>
+        <MessagePopupButtons>
+          <Button onClick={onClose} variant={type === 'error' ? 'stop' : 'start'}>
+            OK
```
[the remark this answers is not in the document]
Comment (frdomovic) [reply src/components/common/MessagePopup.tsx:40]: To not block you with translations, lets update them in later.

File: src-tauri/src/tray/tray.rs [hunk 8]
```diff
@@ -81,6 +82,11 @@ fn add_node_items(menu: SystemTrayMenu, node: &str, is_running: bool) -> SystemT
     } else {
         CustomMenuItem::new(format!("stop_{}", node), "Stop").disabled()
     })
+    .add_item(if is_running {
+        CustomMenuItem::new(format!("dashboard_{}", node), "Dashboard")
+    } else {
+        CustomMenuItem::new(format!("dashboard_{}", node), "Dashboard").disabled()
+    })
```
Comment (miraclx) [inline src-tauri/src/tray/tray.rs:89]: works, but you can avoid duplication

```suggestion
    .add_item({
        let item = CustomMenuItem::new(format!("dashboard_{}", node), "Dashboard");
        is_running.then(|| item).unwrap_or_else(|| item.disabled())
    })
```

**Review by miraclx — APPROVED** [review]

**Pull request merged**